### PR TITLE
assert -> explicit throw

### PR DESCRIPTION
**Pull request by st--**
**Summary**

Replace `@assert` with an explicit `throw()`.

**Follow-up question**

Should we also do the same for the prior consistency checks,
```julia
@assert vfe.fz.f === fx.f
```
could move those into `consistency_check` as well? (NB- `check_consistency` would seem to be a slightly more helpful name, functions-as-verbs, any objections to renaming it? it's meant as an internal function, not exported, so shouldn't break backwards compatibility, or would it?)

### CODE REVIEW

File: src/sparse_approximations.jl [hunk 1]
```diff
@@ -281,7 +281,12 @@ function _compute_intermediates(fx::FiniteGP, y::AbstractVector{<:Real}, fz::Fin
 end
 
 function consistency_check(fx, y)
-    @assert length(fx) == length(y)
+    length(fx) == length(y) || throw(
+        DimensionMismatch(
+            "length(fx) = $(length(fx)) and length(y) = $(length(y)) must match"
```
Comment (devmotion) [inline src/sparse_approximations.jl:286]: Could we use a clearer error message? Probably users won't know what `fx` and `y` are.

**Review by devmotion — CHANGES_REQUESTED** [review]

File: src/sparse_approximations.jl (hunk 1 above)
Comment (st--) [reply src/sparse_approximations.jl:286]: it's the same variable names as the arguments in the docstrings of `elbo` and `dtc`, but how about
```suggestion
            "The FiniteGP's length(fx) = $(length(fx)) and the targets's length(y) = $(length(y)) must match"
```
?

Comment (devmotion) [reply src/sparse_approximations.jl:286]: That's not guaranteed though (we could change them, use the check in other fumctions etc.), and even if the arguments the same names users might not be aware of these names. Therefore I don't think the variable names are helpful at all and suggest removing them completely. Maybe just use something like
```julia
"the dimension of the projected GP ($(length(fx))) and the number of targets ($(length(y))) must be equal"
```

Comment (st--) [reply src/sparse_approximations.jl:286]: If you've got such concrete ideas/thoughts on it, why don't you just go and make it as a code suggestion?

Comment (devmotion) [reply src/sparse_approximations.jl:286]: Actually I wanted to make one but I didn't have time to deal with the inconveniences of the Github app on my phone. It doesn't allow me to add suggestions in follow-up comments via the interface but requires manually copying and modifying the code. I can add a suggestion this evening/night when I have access to my computer.

Comment (st--) [reply src/sparse_approximations.jl:286]: fair enough. how about:
```suggestion
            "the dimension of the projected GP (here: $(length(fx))) must equal the number of targets (here: $(length(y)))"
```

**Pull request merged**